Dockerfile

```diff
@@ -1,18 +1,22 @@
 # --- Builder stage
 FROM node:18-alpine as builder
 WORKDIR /app
-COPY . /app
 
+#Copy package.json first to cache node_modules
+COPY package.json package-lock.json .
 # Scripts include electron-specific dependencies, which we don't need
 RUN npm install --legacy-peer-deps --ignore-scripts
+#Copy code and build with cached modules
+COPY . .
 RUN npm run build:web
 
 # --- Production stage
 FROM nginx:alpine-slim
 
 COPY --chown=nginx:nginx --from=builder /app/release/app/dist/web /usr/share/nginx/html
+COPY ./settings.js.template /settings.js.template
 COPY ng.conf.template /etc/nginx/templates/default.conf.template
 
 ENV PUBLIC_PATH="/"
 EXPOSE 9180
-CMD ["nginx", "-g", "daemon off;"]
+CMD ["/bin/sh", "-c", "envsubst < /settings.js.template > /usr/share/nginx/html/settings.js && envsubst '${PUBLIC_PATH}' < /etc/nginx/templates/default.conf.template > /etc/nginx/conf.d/default.conf && nginx -g 'daemon off;'"]
```

Comment:
The NGINX container already supports envsubst automatically (`NGINX_ENVSUBST_TEMPLATE_DIR`, `/etc/nginx/templates`), maybe that would be cleaner?

Reply:
I did this quite a time ago, but if i remember correctly i was having problems with multiple variables replacing at the same time and leaving a mess so it wouldn't work. This was the solution i found on the spot.

Reply:
Ugh, wrong button press. Sorry for the likely three emails.

You can resolve this by adding something like 
```nginx
  location ${PUBLIC_PATH}settings.js {
    alias /etc/nginx/conf.d/settings.js;
  }
```
to the `ng.conf.template` and then in the Dockerfile doing
```Dockerfile
COPY settings.js.template /etc/nginx/templates/settings.js.template
```

Reply:
I will try, but if i recall correctly it was on the nginx.conf file where it was replacing the envs twice resulting in some gibberish. Will try anyway